packages/autorest.python/autorest/m4reformatter/__init__.py

```diff
@@ -756,7 +758,7 @@ def update_multipart_body_parameter(
             "location": "Method",
             "type": KNOWN_TYPES["anydict"],
             "contentTypes": list(yaml_data.keys()),
-            "defaultContentType": _get_default_content_type(yaml_data.keys()),
+            "defaultContentType": _get_default_content_type(list(yaml_data.keys())),
```

Comment:
nit: I'd prefer if we kept the typing as `Iterable`, then do `len(list(content_types)) == 1` in `_get_default_content_type`. It's best to have all code that can be in helper functions in the helper function